### PR TITLE
Fix/apic 412

### PR DESCRIPTION
**Pull request by carowright**
Bug: Missing query parameters. 

Have to revert previous commit as queryParameters can be of type Object and Array. Now, for hasParameters we check if it's an object or if it's an array and has at least one element. With that, parameters section will be rendered.

### CODE REVIEW

**Comment by jarrodek** [comment]
Why did query parameters became an object? They should be an array of objects. What use case does it covers?

**Comment by carowright** [comment]
I was wrong actually, __computeQueryParameters @return {Array<Object>|Object} Query parameters as object used to be supported. The problem was they were not rendered when defined as queryString because hasParameters did not take into account that it can be an object. 

The difference is between queryParameters and queryString keys. For the first one an array is expected and for the last one only an object containing the values.

File: src/ApiMethodDocumentation.js [hunk 1]
```diff
@@ -654,7 +664,7 @@ export class ApiMethodDocumentation extends AmfHelperMixin(LitElement) {
   /**
    * Computes example headers string for code snippets.
    * @param {Array} headers Headers model from AMF
-   * @return {String|undefined} Computed example value for headers
+   * @return {String|undefind} Computed example value for headers
```
Comment (twoplustwoone) [inline src/ApiMethodDocumentation.js]: Typo `undefined`

File: src/ApiMethodDocumentation.js [hunk 2]
```diff
@@ -409,7 +409,10 @@ export class ApiMethodDocumentation extends AmfHelperMixin(LitElement) {
     if (!methodName || !httpMethod) {
       return true;
     }
-    return methodName.toLowerCase() === httpMethod.toLowerCase();
+    if (methodName.toLowerCase() === httpMethod.toLowerCase()) {
+      return true;
+    }
+    return false;
```
Comment (twoplustwoone) [inline src/ApiMethodDocumentation.js]: Isn't `return methodName.toLowerCase() === httpMethod.toLowerCase();` the same?

File: src/ApiMethodDocumentation.js [hunk 3]
```diff
@@ -671,7 +681,7 @@ export class ApiMethodDocumentation extends AmfHelperMixin(LitElement) {
   /**
    * Computes example payload string for code snippets.
    * @param {Array} payload Payload model from AMF
-   * @return {String|undefined} Computed example value for payload
+   * @return {String|undefind} Computed example value for payload
```
Comment (twoplustwoone) [inline src/ApiMethodDocumentation.js]: Typo `undefined`

**Review by twoplustwoone — APPROVED** [review]

**Comment by jarrodek** [comment]
This still doesn't make sense. Query parameters in the view model is an array an not object. There's never a single query parameter, even with the query string which can be an union. We need here a common interface for all parameters (headers, query params, path params) or otherwise we end up having a different interfaces for each component.

**Pull request merged**